crates/esplora/Cargo.toml

```diff
@@ -21,9 +21,21 @@ futures = { version = "0.3.26", optional = true }
 bitcoin = { version = "0.30.0", optional = true, default-features = false }
 miniscript = { version = "10.0.0", optional = true, default-features = false }
 
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+getrandom = "0.2"
+
+[dev-dependencies]
+electrsd = { version= "0.25.0", features = ["bitcoind_25_0", "esplora_a33e97e1", "legacy"] }
+tokio = { version = "1", features = ["rt", "rt-multi-thread", "macros"] }
+
 [features]
 default = ["std", "async-https", "blocking"]
 std = ["bdk_chain/std"]
 async = ["async-trait", "futures", "esplora-client/async"]
 async-https = ["async", "esplora-client/async-https"]
 blocking = ["esplora-client/blocking"]
+
+# This feature is used to run `cargo check` in our CI targeting wasm. It's not recommended
+# for libraries to explicitly include the "getrandom/js" feature, so we only do it when
+# necessary for running our CI. See: https://docs.rs/getrandom/0.2.8/getrandom/#webassembly-support
+dev-getrandom-wasm = ["getrandom/js"]
```

Comment:
why do we need this. Where do we need to use `getrandom` at all here?

Reply:
It looks like `getrandom` is brought in via the `electrsd` dev-dependency I added. One improvement I can make is to only include `getrandom` as a dev-dependency since that's the only place it's needed.

```
# from esplora directory
$ cargo tree -i getrandom                                                                                                                
getrandom v0.2.10
└── rand_core v0.6.4
    ├── rand v0.8.5
    │   └── secp256k1 v0.27.0
    │       └── bitcoin v0.30.1
    │           ├── bdk_chain v0.5.0 (/Users/steve/git/notmandatory/bdk/crates/chain)
    │           │   └── bdk_esplora v0.3.0 (/Users/steve/git/notmandatory/bdk/crates/esplora)
    │           ├── bitcoincore-rpc-json v0.17.0
    │           │   └── bitcoincore-rpc v0.17.0
    │           │       └── bitcoind v0.33.1
    │           │           └── electrsd v0.25.0
    │           │               [dev-dependencies]
    │           │               └── bdk_esplora v0.3.0 (/Users/steve/git/notmandatory/bdk/crates/esplora)
```

Reply:
I think you don't need it at all since `electrsd` doesn't run in WASM anyway.

Reply:
For some reason `cargo check` with the WASM target is still including `getrandom` and throwing the error. I'll see if I can find some way to force it not to.

Reply:
yeah because it's compiling dev depends in cargo check right? so it's gonna try and compile electrsd for WASM. So you need to `#[cfg(not(target_arch = "wasm32"))]` and not depend on the thing in the Cargo.toml if the arch is WASM.

Tbh I'm not sure why secp256k1 depends on `rand` maybe we can disable a feature that prevents including rand? That would be much simpler.

Reply:
I figured out the issue. Using `[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies]` in the `crates/esplora/Cargo.toml` didn't work until I enabled `resolver = "2"` in the top level workspace `Cargo.toml`. I would have expected it to already be using the new resolver since all our crates are set to `edition = "2021"`, but apparently to resolver also needs to be set for the whole workspace. 

https://blog.rust-lang.org/2021/03/25/Rust-1.51.0.html#cargos-new-feature-resolver